File: Top-Interview-Questions-Leetcode/Easy-Collections/other/number-of-1-bits.cpp
class Solution {
private:
    
    int hammingWeight_recursion(int n) {
        if (n <= 1) return n;
        
        return (n & 1) + hammingWeight_recursion(n >> 1);
    }
    
public:
    int hammingWeight(int n) {
        int cnt = 0;
        
        while (n) {
            if (n & 1) cnt++;
            n >>= 1;
        }
        
        return cnt;
    }
};

File: Top-Interview-Questions-Leetcode/Easy-Collections/other/number-of-1-bits.py
class Solution:
    def __hammingWeight_recursion(self, n: int) -> int:
        if n <= 1:
            return n
        
        return (n & 1) + self.__hammingWeight_recursion(n >> 1)
    
    def hammingWeight(self, n: int) -> int:
        cnt = 0
        
        while n > 0:
            cnt += (n % 2)
            n >>= 1
        
        return cnt